nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -602,13 +611,19 @@ def save_checkpoint(
         # retrieve `sharded_state_dict` if it has not already been configured in `on_save_checkpoint`
         if "sharded_state_dict" not in checkpoint:
             checkpoint["sharded_state_dict"] = self.megatron_parallel.sharded_state_dict()
-        if self.trainer.state.fn == TrainerFn.FITTING and self.ckpt_include_optimizer:
+        if self.trainer.state.fn == TrainerFn.FITTING and self.ckpt_save_optimizer:
             checkpoint["optimizer"] = [self.optimizer_sharded_state_dict()]
 
         self.checkpoint_io.save_checkpoint(checkpoint, filepath, storage_options=storage_options)
 
+    def should_restore_optimizer_states(self, selective_restore: bool = False) -> bool:
+        if selective_restore:
+            return self.selective_restore_config.optimizer_states if self.selective_restore_config else False
```

Comment:
Would that cause any trouble? We are using m argument and a class attribute's value to make a decision. 

Is there any way to make this only use the argument?

Reply:
It's not possible to only use the argument since the helper is called by methods like `load_checkpoint` which are invoked by internal PTL APIs without any args. The arg `selective_restore` just tells the helper whether you're in the selective restore stage or not. Open to suggestions, but couldn't find a better way to accomplish this.